src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -369,7 +369,7 @@ async fn execute(top_match: &clap::ArgMatches) -> Result<(), ExitError> {
                 "Failed to read mtls-client-certificate-chain-path from {cert_chain_path:?}: {err:?}"
               )
             })?;
-            Some(MtlsConfig { key, cert_chain })
+            Some((cert_chain, key))
```

Comment:
A pair of items of the same type, where order matters, makes me uneasy. Especially since we're not consistent with the order (see below). Could this be a struct with named fields, say `cert_chain_pem` and `key_pem`?

Reply:
I agree. We should put this back as a `struct` with two named fields.

Reply:
There's inconsistent feedback; I had them as two fields and was asked to fold them into one option, now it's the other way around. I also did explicitly construct an MtlsConfig from the two pieces, and passed it to TlsConfig. I think "config objects" are quite weird when this local, and what *should* be done is to construct the TlsConfig immediately after loading the data and not in many places.

I believe this is best tackled as part of point 3 here: https://github.com/pantsbuild/pants/issues/19902.